packages/core/useScriptTag/index.ts

```diff
@@ -0,0 +1,153 @@
+import { isString, MaybeRef, tryOnMounted, tryOnUnmounted } from '@vueuse/shared'
+import { ref } from 'vue-demi'
+import { ConfigurableDocument, defaultDocument } from '../_configurable'
+
+export interface UseScriptTagOptions extends ConfigurableDocument {
+  immediate?: boolean
+  async?: boolean
+  type?: string
+  crossOrigin?: 'anonymous' | 'use-credentials'
+  referrerPolicy?: 'no-referrer' | 'no-referrer-when-downgrade' | 'origin' | 'origin-when-cross-origin' | 'same-origin' | 'strict-origin' | 'strict-origin-when-cross-origin' | 'unsafe-url'
+  noModule?: boolean
+  defer?: boolean
+}
+
+/**
+ * Async script tag loading.
+ *
+ * @see   {@link https://vueuse.js.org/useScriptTag}
+ * @param src
+ */
+export function useScriptTag(
+  src: MaybeRef<string>,
+  onLoaded: Function = () => {},
+  {
+    immediate = true,
+    async = true,
+    document = defaultDocument,
+    type = 'text/javascript',
+    crossOrigin,
+    referrerPolicy,
+    noModule,
+    defer,
+  }: UseScriptTagOptions = {},
+) {
+  const scriptTag = ref<HTMLScriptElement>()
```

Comment:
```suggestion
  const scriptTag = ref<HTMLScriptElement | null>(null)
```

I think it's better to have `null` instead of `undefined` here